src/quemb/molbe/chemfrag.py

```diff
@@ -630,17 +727,30 @@ def from_mole(
             is taken from the smaller fragment.
             This means, there will be no centers other than origins.
         """
-        fragments = cls.from_conn_data(
-            mol,
-            BondConnectivity.from_mole(
+        if type(mol) is Mole:
+            fragments = cls.from_conn_data(
                 mol,
-                treat_H_different=treat_H_different,
-                bonds_atoms=bonds_atoms,
-                vdW_radius=vdW_radius,
-            ),
-            n_BE,
-            swallow_replace=swallow_replace,
-        )
+                BondConnectivity.from_mole(
+                    mol,
+                    treat_H_different=treat_H_different,
+                    bonds_atoms=bonds_atoms,
+                    vdW_radius=vdW_radius,
+                ),
+                n_BE,
+                swallow_replace=swallow_replace,
+            )
+        elif type(mol) is Cell:
+            fragments = cls.from_conn_data(
+                mol.to_mol(),
```

Comment:
It would be nice to keep track of `mol` as cell instance.
Otherwise we lose information about lattice vectors etc.

This would require changing the type of `self.mol` to `Mole | Cell` ( as you already do in other occasions)